File: web-app/online-warehouse/src/app/company/dto/company.dto.ts
import {CompanySizeEnum} from "./enum/company-size.enum";
import {ActionTypeEnum} from "./enum/action-type.enum";

export class CompanyDto {
  public id: number;
  public name: string;
  public sizeType: CompanySizeEnum;
  public change: Date;
  public actionType: ActionTypeEnum;

  constructor(
    id: number,
    name: string,
    sizeType: CompanySizeEnum,
    change: Date,
    actionType: ActionTypeEnum
  ) {
    this.id = id;
    this.name = name;
    this.sizeType = sizeType;
    this.change = change;
    this.actionType = actionType;
  }
}

File: core/src/main/java/by/itechart/company/dto/CompanyDto.java
package by.itechart.company.dto;

import by.itechart.company.enums.ActionType;
import by.itechart.company.enums.CompanySize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class CompanyDto {
    private Long id;
    private String name;
    private CompanySize sizeType;
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime change;
    private ActionType actionType;
    private String logo;
}
